ui/components/app/snaps/snap-authorship-pill/snap-authorship-pill.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { Box, IconSize, Text } from '../../../component-library';
+import { SnapIcon } from '../snap-icon';
+import { getSnapMetadata } from '../../../../selectors';
+import {
+  AlignItems,
+  BorderRadius,
+  Display,
+  FlexDirection,
+  TextColor,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+
+type SnapAuthorshipPillProps = {
+  snapId: string;
+  onClick: () => void;
+};
+
+const SnapAuthorshipPill: React.FC<SnapAuthorshipPillProps> = ({
+  snapId,
+  onClick,
+}) => {
+  const { name: snapName } = useSelector((state) =>
+    // @ts-expect-error ts is picking up the wrong type for the selector
+    getSnapMetadata(state, snapId),
+  );
+
+  return (
+    <Box
+      className="snap-authorship-pill"
+      display={Display.Flex}
+      flexDirection={FlexDirection.Row}
+      alignItems={AlignItems.center}
+      borderRadius={BorderRadius.pill}
+      paddingTop={1}
+      paddingBottom={1}
+      paddingLeft={1}
+      paddingRight={2}
+      onClick={onClick}
+    >
+      <SnapIcon avatarSize={IconSize.Sm} snapId={snapId} />
+      <Text
+        color={TextColor.primaryDefault}
+        variant={TextVariant.bodySm}
```

Comment:
Unsure about text variant here, have asked in Slack

Reply:
[01f794e](https://github.com/MetaMask/metamask-extension/pull/26881/commits/01f794e31d0eba1a92ace9f4f266672df814abd2)